packages/expo-modules-core/package.json

```diff
@@ -8,6 +8,26 @@
     "*.fx.ts",
     "./src/polyfill/*.ts"
   ],
+  "exports": {
+    "./package.json": "./package.json",
+    ".": {
+      "types": "./build/index.d.ts",
+      "default": "./src/index.ts"
+    },
+    "./types": "./types.d.ts",
+    "./src/Refs": {
+      "types": "./build/Refs.d.ts",
+      "default": "./src/Refs.ts"
+    },
+    "./src/polyfill/dangerous-internal": {
+      "types": "./build/polyfill/dangerous-internal.d.ts",
+      "default": "./src/polyfill/dangerous-internal.ts"
+    },
+    "./src/uuid/uuid": {
+      "types": "./build/uuid/uuid.d.ts",
+      "default": "./src/uuid/uuid.ts"
+    }
```

Comment:
I think it would be better to pick a more intentional name for the exports, like `"./uuid"` (if it's meant to be public) or `"./private/uuid"`. This way the details around "src" and "build" are hidden and it's clearer whether these are exports are meant to be public or private.

1. Add `"./src/<stuff>"` to exports for compatibility with existing code.
2. Add the more intentionally named exports.
3. Modify import callsites to use the new exports.
4. Remove `"./src/<stuff>"`

Reply:
agree with the public api stuff.

was thinking if we have the `expo-modules-core/src/*`. non from us but some from externals:
https://github.com/search?type=code&q=%2Fimport.*expo-modules-core%5C%2Fsrc%2F
we may need to further add `src/PermissionsInterface`
https://github.com/Gepuy/SimpleCameraApp/blob/de4ca692aa36e543139cb51db1e0d73db8d43947/src/screens/CameraScreen.tsx#L6

Reply:
This would be yet another breaking change in core. It could always come later.